src/coreclr/gc/gcinterface.h

```diff
@@ -485,64 +485,68 @@ typedef void (* fq_scan_fn)(Object** ppObject, ScanContext *pSC, uint32_t dwFlag
 typedef void (* handle_scan_fn)(Object** pRef, Object* pSec, uint32_t flags, ScanContext* context, bool isDependent);
 typedef bool (* async_pin_enum_fn)(Object* object, void* context);
 
-
+#if defined(FEATURE_NATIVEAOT) && !defined(TARGET_WINDOWS)
+#define PURE_VIRTUAL { assert(!"pure virtual function called"); while(true); }
```

Comment:
`while(true)` looks odd here, is there any precedent where its done for similar cases?

Reply:
It can use a comment on what it is trying to achieve.

We have a similar infinite loop for example here: https://github.com/dotnet/runtime/blob/9d6396deb02161f5ee47af72ccac52c2e1bae458/src/coreclr/vm/fcall.h#L814 (it is done for a different purpose, but it would be very confusing without the comment).

Reply:
This is an infinite spin so we can find issues faster in release builds. I agree this could be annoying. I'm fine with it as is, but perhaps we could instead insert a `FAILFAST`? @jkotas or @am11 Any push back?

Reply:
I would keep this as simple as possible. The infinite loop is here since we do not have a good value to return. The chance that somebody will hit it in release build without seeing the asset in debug builds before are close to zero.

Also, it won't be an infinite loop in release build, when compiled with recent clang at least. It will get optimized into nothing (yes, C/C++ standard allows that).

Reply:
> It will get optimized into nothing (yes, C/C++ standard allows that)

Indeed https://github.com/llvm/llvm-project/blob/f5cbbb998826bc84cbebd12f890622b3ec3184fe/llvm/lib/Transforms/IPO/WholeProgramDevirt.cpp#L1018-L1021 😅

Reply:
a comment would certainly help to reduce confusion for the reader here :)